src/Types/StructureType.php

```diff
@@ -48,10 +49,10 @@ public function addField($field, Type $type): void
 	/**
 	 * @param int|string $field
 	 */
-	public function overwriteInvalidField($field, Type $type): void
+	public function overwriteInvalidField($field, WithTypeAndValue $typeAndValue): void
 	{
-		$this->fields[$field] = $type;
-		$this->invalidFields[$field] = true;
+		$this->fields[$field] = $typeAndValue->getInvalidType();
```

Comment:
Change name to `setInvalidField` and drop the override of fields please. It's not needed anymore, formatter can use getInvalidFields() instead

Reply:
We will have to figure out how to merge `fields` and `invalidField`. Because formatter needs both.
https://github.com/orisai/object-mapper/blob/c5a9db548383d59baae8b9b77a0f64a3299c2f41/src/Formatting/VisualErrorFormatter.php#L229-L244

Reply:
Just prefer invalid, if they exist. If all should be rendered, merge the arrays (prioritize the invalid), keys are field names.

Reply:
I'd say that `filterFields` should return `array<WithTypeAndValue>` insted of `array<Type>`. 
I am not able to get the value of `field` (cos it's `Type`), cos they are not tracked. Values are tracked only for `invalidFields`.